crates/sui-transaction-builder/src/lib.rs

```diff
@@ -0,0 +1,917 @@
+
+impl From<Value> for Input {
+    fn from(value: Value) -> Input {
+        match value {
+            Value::Input(i) => Input::Pure(bcs::to_bytes(&i.to_be_bytes()).unwrap()),
+            _ => panic!("Cannot convert Value to Input"),
+        }
+    }
+}
+
+// this seems to be needed as without it this is not possible
+// let input = tx.input(Serialized(&1u64).serialize());
+impl From<Vec<u8>> for Input {
+    fn from(bytes: Vec<u8>) -> Input {
+        Input::Pure(bytes)
+    }
+}
+
+impl<'a, T: Serialize> Serialized<'a, T> {
+    /// Serializes the given value into bcs bytes.
+    pub fn serialize(&self) -> Vec<u8> {
+        bcs::to_bytes(self.0).unwrap()
+    }
+}
+
+// Allow converting Serialized<Address> into Input
+impl<'a> From<Serialized<'a, Address>> for Input {
+    fn from(serialized: Serialized<'a, Address>) -> Input {
+        // Convert Serialized<Address> into Input::Pure variant
+        Input::Pure(serialized.serialize()) // Here we use `Pure(Vec<u8>)`
+    }
+}
+
+/// Convert a [`Value`] into a transaction command [`Argument`] type.
+impl From<Value> for Argument {
+    fn from(value: Value) -> Self {
+        match value {
+            Value::Gas => Argument::Gas,
+            Value::Input(i) => Argument::Input(i),
+            Value::Result(i) => Argument::Result(i),
+            Value::NestedResult(i, j) => Argument::NestedResult(i, j),
+        }
+    }
+}
+
+impl TryFrom<&Input> for InputArgument {
+    type Error = Error;
+    fn try_from(value: &Input) -> Result<Self, Error> {
+        match value {
+            Input::Object(ref object) => {
+                if let Some(obj) = &object.kind {
+                    match obj {
+                        Kind::ImmOrOwned => Ok(InputArgument::ImmutableOrOwned(object.try_into()?)),
+                        Kind::Receiving => Ok(InputArgument::Receiving(object.try_into()?)),
+                        Kind::Shared => Ok(InputArgument::Shared {
+                            object_id: object.id,
+                            initial_shared_version: object
+                                .initial_shared_version
+                                .ok_or_else(|| Error::msg("Initial shared version not found"))?,
+                            mutable: object.mutable.ok_or_else(|| {
+                                Error::msg("Expected mutable object, but mutable field is None")
+                            })?,
+                        }),
+                    }
+                } else {
+                    Err(anyhow!("Object kind not found"))
+                }
+            }
+
+            Input::Pure(v) => Ok(InputArgument::Pure { value: v.clone() }),
+        }
+    }
+}
+
+/// Convert an object into an [`UnresolvedObjectReference`].
+fn to_unresolved_obj_ref(obj: Object) -> UnresolvedObjectReference {
+    UnresolvedObjectReference {
+        object_id: obj.id,
+        version: obj.version,
+        digest: obj.digest,
+    }
+}
+
+impl From<Input> for UnresolvedInputArgument {
```

Comment:
Similar question as for `Value` and `Argument` on whether these types can be merged -- they are intended to represent the same domain, so it makes sense that you can implement conversions between them, but maybe we can avoid that entirely.

Reply:
I am thinking how we can do it, but what about `Literals`, when we get tx resolution?

Reply:
Literals are specific to transaction building, there is not an analogous type in the types crate, but that's fine because we just implement `From<Literal> for Input` (or whatever I suggested we should use instead of `Input` above).

Reply:
I implemented this as well, except the literal part. Tests pass but I am not 100% confident it's correct.